rcl_yaml_param_parser/test/test_parser.cpp

```diff
@@ -0,0 +1,351 @@
+    params_st->allocator, num_malloc_calls_until_copy_param + 1);
+  EXPECT_EQ(nullptr, rcl_yaml_node_struct_copy(params_st));
+
+  // Check bool_array_value->values is set to null if size is 0
+  set_time_bomb_allocator_malloc_count(params_st->allocator, -1);
+  params_st->params->parameter_values[0].bool_array_value->size = 0u;
+  copy = rcl_yaml_node_struct_copy(params_st);
+  EXPECT_NE(nullptr, copy);
+  EXPECT_EQ(nullptr, copy->params->parameter_values[0].bool_array_value->values);
+  rcl_yaml_node_struct_fini(copy);
+
+  // Check integer_array array pointer is allocated
+  int64_t integer_array[] = {42};
+  rcl_int64_array_s temp_integer_array = {&integer_array[0], 1};
+  set_time_bomb_allocator_malloc_count(params_st->allocator, num_malloc_calls_until_copy_param);
+  params_st->params->parameter_values[0].bool_array_value = nullptr;
+  params_st->params->parameter_values[0].integer_array_value = &temp_integer_array;
+  EXPECT_EQ(nullptr, rcl_yaml_node_struct_copy(params_st));
+
+  // Check integer_array->values is allocated
+  set_time_bomb_allocator_malloc_count(
+    params_st->allocator, num_malloc_calls_until_copy_param + 1);
+  EXPECT_EQ(nullptr, rcl_yaml_node_struct_copy(params_st));
+
+  // Check integer_array->values is set to null if size is 0
+  params_st->params->parameter_values[0].integer_array_value->size = 0u;
+  copy = rcl_yaml_node_struct_copy(params_st);
+  EXPECT_NE(nullptr, copy);
+  EXPECT_EQ(nullptr, copy->params->parameter_values[0].integer_array_value->values);
+  rcl_yaml_node_struct_fini(copy);
+
+
+  double double_array[] = {42.0};
+  rcl_double_array_s temp_double_array = {&double_array[0], 1};
+  set_time_bomb_allocator_malloc_count(params_st->allocator, num_malloc_calls_until_copy_param);
+  params_st->params->parameter_values[0].integer_array_value = nullptr;
+  params_st->params->parameter_values[0].double_array_value = &temp_double_array;
+  EXPECT_EQ(nullptr, rcl_yaml_node_struct_copy(params_st));
+
+  set_time_bomb_allocator_malloc_count(
+    params_st->allocator, num_malloc_calls_until_copy_param + 1);
+  EXPECT_EQ(nullptr, rcl_yaml_node_struct_copy(params_st));
+
+  params_st->params->parameter_values[0].double_array_value->size = 0u;
+  copy = rcl_yaml_node_struct_copy(params_st);
+  EXPECT_NE(nullptr, copy);
+  EXPECT_EQ(nullptr, copy->params->parameter_values[0].double_array_value->values);
+  rcl_yaml_node_struct_fini(copy);
+
+  char s[] = "stringy string";
+  char * string_array[] = {&s[0]};
+  rcutils_string_array_t temp_string_array = {1, &string_array[0], allocator};
+  set_time_bomb_allocator_malloc_count(params_st->allocator, num_malloc_calls_until_copy_param);
+  params_st->params->parameter_values[0].double_array_value = nullptr;
+  params_st->params->parameter_values[0].string_array_value = &temp_string_array;
+  EXPECT_EQ(nullptr, rcl_yaml_node_struct_copy(params_st));
+
+  for (int i = 0; i < 5; ++i) {
+    set_time_bomb_allocator_calloc_count(params_st->allocator, 4);
```

Comment:
Why are you doing this 5 times? or:

```suggestion
    set_time_bomb_allocator_calloc_count(params_st->allocator, i);
```

Reply:
Fixed.